### PR TITLE
fix(jdbc-sink): generate insert into stmt for append-only sink

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
- Generate insert into stmt for append-only sink

Fix https://github.com/risingwavelabs/risingwave/issues/10924

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/connector/src/sink/remote.rs [hunk 1]
```diff
@@ -231,7 +231,7 @@ impl RemoteSinkWriter {
                     &config.connector_type, e
                 )
             })?;
-        tracing::info!(
+        tracing::trace!(
```
Comment (wenym1) [inline src/connector/src/sink/remote.rs:234]: Any reason to make it trace? It is not expected to have too many these logs.

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java [hunk 2]
```diff
@@ -111,23 +120,30 @@ private static List<String> getPkColumnNames(
         return pkColumnNames;
     }
 
-    private PreparedStatement prepareUpsertStatement(SinkRow row) {
+    private PreparedStatement prepareInsertOrUpsertStatement(SinkRow row) {
         try {
+            var preparedStmt = config.isUpsertSink() ? upsertPreparedStmt : insertPreparedStmt;
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java:125]: Here we implicitly introduce an assumption that the `upsertPreparedStmt` has the same number of argument as `insertPreparedStmt`. 

We'd better keep the original method and implement another method `prepareInsertStatement` specially for append only sink.

File: java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/jdbc/JdbcDialect.java [hunk 3]
```diff
@@ -61,7 +61,7 @@ public interface JdbcDialect {
     Optional<String> getUpsertStatement(
             SchemaTableName schemaTableName, List<String> fieldNames, List<String> uniqueKeyFields);
 
-    void bindUpsertStatement(
+    void bindInsertOrUpsertStatement(
```
Comment (wenym1) [inline java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/jdbc/JdbcDialect.java:64]: `upsert` means `insert or update`. I think we don't need to change the name.

To handle insert for append only sink, we can simply call the `bindInsertStatement`.

**Review by lmatz — APPROVED** [review]
LGTM for the test cases

**Review by wenym1 — APPROVED** [review]

**Pull request merged**